include/boost/geometry/index/detail/rtree/query_iterators.hpp

```diff
@@ -294,12 +294,6 @@ class query_iterator
         return *this;
     }
 
-    query_iterator(query_iterator && o)
```

Comment:
@awulkiew this is just my quick fix to make tests passing.

Reply:
@vissarion I don't think that this is necessary. Which test doesn't pass?

Reply:
Right, I can reproduce it with gcc-6. It is caused by the replacement of `scoped_ptr` with `unique_ptr`. So direct solution would be to pass `nullptr` instead.

But the solution you prosed is even better (almost). AFAIU your intention was to force the compiler to generate move ctor automatically. But since copy ctor and assignements are defined AFAIR this will not happen. Instead copy ctor will be called which will harm the performance. Instead move ctor should be declared as `default`. And since we're at it I think move assignment operator could be `default` as well, hence:
```
query_iterator(query_iterator&&) = default;
query_iterator& operator=(query_iterator &&) = default;
```

Reply:
Sure, why not. Do we still need the current PR. Either we close it and you fix the issues there or we merge it and you create a PR with additional fixes. I am OK with both. 

Also there are some tests related to envelop that are failing but I want to fix them in a separate PR.

Reply:
I propose to merge this PR.